analyzers/src/SonarAnalyzer.CSharp/Rules/EmptyNestedBlock.cs

```diff
@@ -32,50 +31,8 @@ namespace SonarAnalyzer.Rules.CSharp
 {
     [DiagnosticAnalyzer(LanguageNames.CSharp)]
     [Rule(DiagnosticId)]
-    public sealed class EmptyNestedBlock : EmptyNestedBlockBase
+    public sealed class EmptyNestedBlock : EmptyNestedBlockBase<SyntaxKind>
     {
-        private static readonly DiagnosticDescriptor rule =
-            DiagnosticDescriptorBuilder.GetDescriptor(DiagnosticId, MessageFormat, RspecStrings.ResourceManager);
-
-        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; } = ImmutableArray.Create(rule);
-
-        protected override void Initialize(SonarAnalysisContext context)
-        {
-            context.RegisterSyntaxNodeActionInNonGenerated(
-                c =>
-                {
-                    if (IsEmpty(c.Node))
-                    {
-                        c.ReportDiagnosticWhenActive(Diagnostic.Create(rule, c.Node.GetLocation()));
-                    }
-                },
-                SyntaxKind.Block,
-                SyntaxKind.SwitchStatement);
-        }
-
-        private static bool IsEmpty(SyntaxNode node)
-        {
-            var blockNode = node as BlockSyntax;
-
-            return (node is SwitchStatementSyntax switchNode && IsEmpty(switchNode)) ||
-                (blockNode != null && IsNestedAndEmpty(blockNode));
-        }
-
-        private static bool IsEmpty(SwitchStatementSyntax node)
-        {
-            return !node.Sections.Any();
-        }
-
-        private static bool IsNestedAndEmpty(BlockSyntax node)
-        {
-            return IsNested(node) && IsEmpty(node);
-        }
-
-        private static bool IsNested(BlockSyntax node)
-        {
-            return !AllowedContainerKinds.Contains(node.Parent.Kind());
-        }
-
         private static readonly ISet<SyntaxKind> AllowedContainerKinds = new HashSet<SyntaxKind>
         {
```

Comment:
An array of 2 items is faster then an hashset.

Reply:
fair point, although it's six items and you also need to replace the `.Contains` with a loop to do the equality checks

Reply:
Well, the hash of an enum, will not win for 6 items in an array. It will not harm either.